docs/eventing/experimental-features.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "Experimental features"
+weight: 99
+type: "docs"
+---
+
+# Experimental features
+
+In Knative Eventing we want to keep the innovation alive, experimenting and delivering new features without affecting the stability of the project.
+
+In order to achieve that goal, we have a process to include new features.
+This allows users like you to try out new features and provide feedback back to the project.
+
+This document explains how to enable experimental features and which ones are available today.
+
+For more details about the process, the feature phases, quality requirements and guarantees, check out the [Experimental features process documentation](https://github.com/knative/eventing/blob/main/docs/experimental-features.md).
+
+## Before you begin
+
+You must have a Knative cluster running with the Eventing component installed. [Learn more](../install/)
+
+## Experimental features configuration
+
+When installing Eventing, the `config-experimental-features` ConfigMap is added to your cluster in the `knative-eventing` namespace.
+In order to enable a feature, you just need to add it to the config map and set its value to `"true"`.
+For example, to enable `new-cool-feature`:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-experimental-features
+  namespace: knative-eventing
+  labels:
+    eventing.knative.dev/release: devel
+    knative.dev/config-propagation: original
+    knative.dev/config-category: eventing
+data:
+  new-cool-feature: "true"
+```
+
+In order to disable it, you can either remove the flag or set it to `"false"`:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-experimental-features
+  namespace: knative-eventing
+  labels:
+    eventing.knative.dev/release: devel
+    knative.dev/config-propagation: original
+    knative.dev/config-category: eventing
+data:
+  new-cool-feature: "false"
```

Comment:
I guess this is really a comment on the eventing implementation of this rather than the doc, but Im only noticing this here now :).. Would it make sense (I think the proposal doc said we were going to) to align this with the serving features configmap where the settings are "enabled", "disabled", "allowed"?

Reply:
We talked about that, but we decided to avoid having the opt-in (see the mail thread) features in that same CM (like you have in serving), hence this CM is only about "experimental feature on/off".

In any case, feel free to reply to that mail thread if you have some concrete proposal around it :). Can we ignore this discussion here for the sake of this PR?

Reply:
FWIW I think some of the serving features also only have "enabled" & "disabled", so I don't _think_ there's a reason we can't align, even if you don't have a need for an "allowed" state yet. Anyway let me go look for the email and reply there, I didn't realise there was a mail thread going on as well as the proposal doc (I've been filtering out stuff tagged [Eventing])

Reply:
The original doc it had the following clause - I'm surprised after the discussion in the mailing list the doc wasn't updated and comments resolved.

> We store in a config map a map of enabled/disabled features. Pretty much the same as Kubernetes Feature Gates or https://github.com/knative/serving/blob/master/pkg/apis/config/features.go

The tri-state is more expressive and allows for _users_ to individually opt into experimental behaviour - in contrast to the operator enabling it for everyone.

This allows for a gradual rollout of features.

Reply:
> The tri-state is more expressive and allows for users to individually opt into experimental behaviour - in contrast to the operator enabling it for everyone.

This was discussed as well both in the doc and in the mailing list: depending on the feature, we're going to decide if we want it opt-in and opt-out by default, but because eventing has plenty of CMs, most of them already namespace scoped, it doesn't make sense to have this ConfigMap too for enabling component features.